apps/scan/backend/src/electrical_testing/server.ts

```diff
@@ -4,14 +4,70 @@ import { PORT } from '../globals';
 import { buildApp } from './app';
 import { cardReadAndUsbDriveWriteLoop, printAndScanLoop } from './background';
 import { ServerContext } from './context';
+import {
+  createSimpleScannerClient,
+  SimpleScannerClient,
+} from './simple_scanner_client';
+
+export interface ElectricalTestingServerContext extends ServerContext {
+  scannerClient: SimpleScannerClient;
+  controller: AbortController;
+}
 
 export function startElectricalTestingServer(context: ServerContext): void {
   const { logger } = context;
+  const controller = new AbortController();
+  const client = createSimpleScannerClient();
+  const testContext: ElectricalTestingServerContext = {
+    ...context,
+    scannerClient: client,
+    controller,
+  };
+  const cardReadAndUsbDriveWriteLoopPromise =
+    cardReadAndUsbDriveWriteLoop(testContext);
+  const printAndScanLoopPromise = printAndScanLoop(testContext);
 
-  setTimeout(() => cardReadAndUsbDriveWriteLoop(context));
-  setTimeout(() => printAndScanLoop());
+  void logger.log(LogEventId.BackgroundTaskStarted, 'system', {
+    disposition: 'success',
+    message: 'Starting card read loop',
+  });
+
+  void logger.log(LogEventId.BackgroundTaskStarted, 'system', {
+    disposition: 'success',
+    message: 'Starting print and scan loop',
+  });
+
+  cardReadAndUsbDriveWriteLoopPromise
+    .then(() => {
+      void logger.log(LogEventId.BackgroundTaskCompleted, 'system', {
+        disposition: 'success',
+        message: 'Card read and USB drive write loop completed',
+      });
```

Comment:
To make sure I follow, this would only get emitted if someone clicks the stop test button, correct?

Reply:
Yep, that's right.